projects/storefrontlib/src/lib/global-message/facade/global-message.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import * as fromStore from '../store';
+import { GlobalMessage, GlobalMessageType } from '../models/message.model';
+
+@Injectable()
+export class GlobalMessageService {
+  readonly messages$: Observable<any> = this.store.pipe(
+    select(fromStore.getGlobalMessagesEntities),
+    filter(data => data !== undefined)
+  );
+
+  constructor(private store: Store<fromStore.GlobalMessageState>) {}
+
+  add(message: GlobalMessage) {
+    this.store.dispatch(new fromStore.AddMessage(message));
+  }
+
+  remove(type: GlobalMessageType, index: number) {
+    this.store.dispatch(
+      new fromStore.RemoveMessage({
+        type: type,
+        index: index
+      })
+    );
+  }
+
+  removeByType(type: GlobalMessageType) {
+    this.store.dispatch(new fromStore.RemoveMessagesByType(type));
+  }
```

Comment:
Do you think it would be better if we overloaded the existing remove method?
We would have:
```
remove(type: GlobalMessageType, index: number) {...}
remove(type: GlobalMessageType) {...}
```

Reply:
yes, we can just one have method "remove". index is optional. If no index passed, we remove the whole type of messages.